_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+
+## Kafka Connector
+
+Kafka Connector plugin is the solution to your problem when it comes to integrating OpenFaaS as a component of a bigger distributed system which is using Kafka to make communication between its parts. You just need to apply annotation with the topic you would like to subscribe your function to and then run the connector with your Kafka configuration.
+
+Here is a diagram which shows how the component works:
+
+![Form Options](/images/kafka-connector/overview.jpg)
+
+## Pre-requisites
+
+First and most important thing is [running OpenFaaS instance](https://docs.openfaas.com/deployment/kubernetes/) with default configuration on Kubernetes, for our case [minikube deployment](https://kubernetes.io/docs/setup/minikube/) is more than enough.
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone and configure Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector.git
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+The file `connector-dep.yaml` contains all the configuration you need to set for your connector.
+
+It looks something like:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: kafka
+    component: kafka-connector
+  name: kafka-connector
+  namespace: openfaas
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kafka
+        component: kafka-connector
+    spec:
+      containers:
+      - name: kafka
+        image: openfaas/kafka-connector:0.4.0
+        env:
+          - name: gateway_url
+            value: "http://gateway.openfaas:8080"
+          - name: topics
+            value: "faas-request,"
+          - name: print_response
+            value: "true"
+          - name: print_response_body
+            value: "true"
+          - name: basic_auth
+            value: "true"
+          - name: secret_mount_path
+            value: "/var/secrets/"
+          - name: kafka_broker
+            value: "kafka"
+        volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+      volumes:
+      - name: auth
+        secret:
+          secretName: basic-auth
+```
+
+If you deployed your OpenFaaS instance with Helm and default values there is no need to make changes to this file. The `kafka_broker` variable is our broker host from which we will pull topics.
```

Comment:
We need to make sure people know how to connect to their existing broker including use of the namespace.

Reply:
Added comment that if their kafka service is in another namespace they can append `.<namespace>` to make the broker discoverable by the connector 💯